code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -175,38 +175,73 @@
-	set desc = "Send a telepathic message to an unlucky recipient."
+	set desc = "Send a telepathic message to a recipient."
+
+	var/obj/item/organ/external/rhand = src.get_organ("r_hand")
+	var/obj/item/organ/external/lhand = src.get_organ("l_hand")
+	if((!rhand || !rhand.is_usable()) && (!lhand || !lhand.is_usable()))
+		src <<"<span class='warning'>You can't communicate without the ability to use your hands!</span>"
+		return FALSE
+	if(stat || paralysis || stunned || weakened ||  restrained())
+		src <<"<span class='warning'>You can't communicate while unable to move your hands to your head!</span>"
+		return FALSE
+	if(last_special > world.time)
+		src << "<span class='notice'>Your mind requires rest!</span>"
+		return FALSE
+
+	last_special = world.time + 40
+
+	visible_message("<span class='notice'>[src] touches their fingers to their temple.</span>")
 
 	var/list/targets = list()
-	var/target = null
+	for(var/mob/living/M in view(client.view, client.eye))
+		targets += M
+	var/mob/living/target = null
 	var/text = null
 
-	targets += getmobs() //Fill list, prompt user with list
 	target = input("Select a creature!", "Speak to creature", null, null) as null|anything in targets
 
-	if(!target) return
+	if(!target)
+		return
 
 	text = input("What would you like to say?", "Speak to creature", null, null)
 
 	text = sanitize(text)
 
-	if(!text) return
+	if(!text)
+		return
 
-	var/mob/M = targets[target]
+	if(istype(target, /mob/abstract/observer) || target.stat == DEAD)
+		to_chat(src, "<span class='warning'>Not even a [src.species.name] can speak to the dead.</span>")
+		return
 
-	if(istype(M, /mob/abstract/observer) || M.stat == DEAD)
-		src << "Not even a [src.species.name] can speak to the dead."
+	if (target.isSynthetic())
+		to_chat(src, "<span class='warning'>This can only be used on living organisms.</span>")
 		return
 
-	log_say("[key_name(src)] communed to [key_name(M)]: [text]",ckey=key_name(src))
 
-	M << "<span class='notice'>Like lead slabs crashing into the ocean, alien thoughts drop into your mind: [text]</span>"
-	if(istype(M,/mob/living/carbon/human))
-		var/mob/living/carbon/human/H = M
-		if(H.species.name == src.species.name)
-			return
-		H << "<span class='warning'>Your nose begins to bleed...</span>"
-		H.drip(1)
+	if (target.is_diona())
+		to_chat(src, "<span class='alium'>The creature's mind is not solid enough and slips through like sand.</span>")
+		return
+
+	if(!(target in view(client.view, client.eye)))
+		to_chat(src, "<span class='warning'>[target] is too far for your mind to grasp!</span>")
+		return
+
+	log_say("[key_name(src)] communed to [key_name(target)]: [text]",ckey=key_name(src))
+
+	var/mob/living/carbon/human/H = target
+	if (/mob/living/carbon/human/proc/commune in target.verbs)
+		target << "<span class='psychic'>You instinctively sense [src] sending their thoughts into your mind, hearing:</span> [text]"
+	else if(prob(25) && (target.mind && target.mind.assigned_role=="Chaplain"))
```

Comment:
Why just the chaplain ?

Wouldn't it make more sense to base it on the target species ?
i.e. Humans having a slightly higher chance than maybe Unathi to figure out who talked to them ?

Reply:
Because while none of those races are inherently any psychically attuned unless viruses fuck with them, we have established that the Chaplain (and many priests across the galaxy, notably the Tribunal Priests) is attuned to that kind of thing.